.github/workflows/wheels-test.yaml

```
@@ -0,0 +1,147 @@
+        type: string
+        default: "ci/test_wheel.sh"
+
+      # general settings
+      matrix_filter:
+        type: string
+        default: "."
+
+      # test settings
+      test-docker-options:
+        required: false
+        type: string
+        default: '-e _NOOP'
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  discussions: none
+  id-token: write
+  issues: none
+  packages: read
+  pages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    env:
+      BUILD_TYPE: ${{ inputs.build_type }}
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Validate test type
+        run: |
+          if [[ "$BUILD_TYPE" != "pull-request" ]] && [[ "$BUILD_TYPE" != "nightly" ]]; then
+              echo "Invalid build type! Must be 'nightly' or 'pull-request'."
+              exit 1
+          fi
+      - name: Compute test matrix
+        id: compute-matrix
+        env:
+          SMOKE_TEST_CMD: ${{ inputs.test-smoketest }}
+          UNIT_TEST_CMD: ${{ inputs.test-unittest }}
+        run: |
+          set -eo pipefail
+
+          export MATRICES="
+            pull-request:
+              - { arch: 'amd64', python: '3.9', ctk: '11.8.0', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'amd64', python: '3.10', ctk: '12.0.1', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'arm64', python: '3.9', ctk: '12.0.1', image: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+            nightly:
+              - { arch: 'amd64', python: '3.9', ctk: '11.8.0', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'amd64', python: '3.10', ctk: '11.8.0', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'arm64', python: '3.9', ctk: '11.8.0', image: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+              - { arch: 'arm64', python: '3.10', ctk: '11.8.0', image: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+              - { arch: 'amd64', python: '3.9', ctk: '12.0.1', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'amd64', python: '3.10', ctk: '12.0.1', image: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
+              - { arch: 'arm64', python: '3.9', ctk: '12.0.1', image: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+              - { arch: 'arm64', python: '3.10', ctk: '12.0.1', image: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
+          "
+
+          TEST_MATRIX=$(yq -n 'env(MATRICES) | .[strenv(BUILD_TYPE)]')
+          export TEST_MATRIX
+
+          echo "MATRIX=$(
+            yq -n -o json 'env(TEST_MATRIX)' | \
+            jq -c '${{ inputs.matrix_filter }} | {include: .}' \
+          )" | tee --append "${GITHUB_OUTPUT}"
+
+  wheel-test:
```

```suggestion
  tests:
```
Similar to https://github.com/rapidsai/shared-action-workflows/blob/016436cbac2d92f0f0b4f78a63262117068a8aee/.github/workflows/conda-cpp-tests.yaml#L99